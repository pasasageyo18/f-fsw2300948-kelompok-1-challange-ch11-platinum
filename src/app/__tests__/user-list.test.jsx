import Page from '../user-list/page'
import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import { jest } from '@jest/globals'


jest.mock('firebase/firestore', () => ({
  collection: jest.fn(),
  getDocs: jest.fn(),
}))

describe('should render Users List', () => {
  it('render user list', () => {
    render(<Page />)
    expect(screen.getByTestId('user-list')).toBeInTheDocument()
  })

  it('should be the same component', () => {
    render(<Page />)
    const userList = screen.getByTestId('user-list')
    expect(userList).toMatchSnapshot()
  })

  it('should be null', () => {
    render(<Page />)
    const userName = screen.getByTestId('user-name')
    expect(userName).toBeInTheDocument()
  })
  it('should render users info', async () => {
    render(<Page />)
    const userInfo = screen.getByTestId('user-info')
    expect(userInfo).toBeGreaterThan(1)
  })

  it('should render link', () => {
    render(<Page />)
    const link = screen.getByTestId('link')
    expect(link).toBeInTheDocument()
  })
})
