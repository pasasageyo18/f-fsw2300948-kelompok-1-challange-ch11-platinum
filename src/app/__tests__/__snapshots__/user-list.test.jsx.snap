// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Users List should be the same component 1`] = `
<div
  class="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 gap-4"
  data-testid="user-list"
/>
`;
